### PR TITLE
Containerized Dev Environment

### PR DESCRIPTION
**Pull request by sam-aldis**
[issue](https://github.com/OWASP/SecureTea-Project/issues/241)

## Status
**READY/IN DEVELOPMENT/HOLD**

## Description
Added development Dockerfile with python3, nodejs and a code server running inside it.

## Todos
- [x] Tests
- [-] Documentation

## Steps to Test or Reproduce
See updated README.md

## Impacted Areas in Application
List general components of the application that this PR will affect:

Development Experience
Containerization

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -90,6 +90,22 @@ For more detailed information, refer to the [installation guide](/doc/en-US/user
 
 For more detailed information, refer to the [usage guide](https://github.com/OWASP/SecureTea-Project/blob/master/doc/en-US/user_guide.md#usage).
 
+## Quick Start Developer
```
Comment (rejahrehim) [inline README.md]: Can we move this section to the developer docs?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+	tar --gzip -xf ./code-server2.preview.11-vsc1.37.0-linux-x86_64.tar.gz && \
+	mv -f code-server2.preview.11-vsc1.37.0-linux-x86_64/* ~/ && \
+	chmod +x ~/code-server
+RUN apt install zsh gcc clang libnl-utils libnfnetlink-dev libnfnetlink0 libnl-cli-3-dev libnl-3-dev -y
+RUN apt install -y build-essential python-dev libnetfilter-queue-dev
+RUN apt -y install clamav 
+RUN freshclam
+RUN npm i -g @angular/cli typescript tslib tslint serve webpack parcel ts-node
+WORKDIR /root/securetea
+RUN pip install virtualenv && \
```
[the remark this answers is not in the document]
Comment (sam-aldis) [reply Dockerfile]: ignoring freezing virtualenv from pip

**Comment by sam-aldis** [comment]
virtualenv isn't fixed in the docker file intentionally as it will allow for
future updates and bug fixes to be pulled into the image!

**Comment by rejahrehim** [comment]
@all-contributors please add @sam-aldis  code

**Comment by sam-aldis** [comment]
Thanks Rejah,
    Still have to work out if I'm going to need to compile the required
libraries from
scratch for the docker image :') can I ask quickly what distro your running
on as
I could always move the docker image over to that. It's ubuntu purely
because it
seems to be the most popular distro for developers at the moment.

Samuel Aldis

**Comment by abhisharma404** [comment]
@sam-aldis Hi there, this project is primarily being developed and tested on Ubuntu 18.04 LTS, though it extends supports for all Debian based OS, like Kali Linux (all versions) and all versions of Ubuntu.

**Pull request merged**